include/mbedtls/platform_util.h

```diff
@@ -41,6 +41,39 @@
 extern "C" {
 #endif
 
+#if defined(MBEDTLS_PARAM_FAILED_CALLBACK)
+
+/**
+ * \brief       Callback function for parameter validation failure.
+ *
+ *              When the MBEDTLS_CHECK_PARAMS option is enabled, the library
+ *              provides additional validation of all input parameters to
+ *              confirm that they conform to what the interface can accept.
+ *              For example - NULL paramater checks.
+ *
+ *              These checks are designed to check programmatic issues in the
+ *              application software using Mbed TLS, or catch other runtime
+ *              errors which may be due to issues in the application software.
+ *
+ *              The function this points to will be called upon parameter
+ *              validation failure if the MBEDTLS_PARAM_FAILED_CALLBACK option
+ *              has been enabled.
+ */
+extern void (*mbedtls_param_failed)( const char *, const char *, int );
```

Comment:
I am not sure about this. While I find it reasonable, it is not a common practice in the library (e.g. `nv_seed_{read|write}` are stored and set using explicit type definitions. Additionally, added to the public header, such a typedef would have to be documented and become a part of the public API. Since it is not useful outside the library, I don't see much gain from doing so. Could you please explain the advantage of extracting the type to a typedef?